### PR TITLE
Add creating pages doc

### PR DESCRIPTION
**Pull request by calcsam**

### CODE REVIEW

File: docs/docs/how-to/local-development/javascript-tooling.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+title: Javascript tooling
+---
+
+Gatsby offers support for the standard Javascript toolchain.
+
+## Typescript
+
+Gatsby supports Typescript automatically, with the [ability to modify the default configuration](gatsby-plugin-typescript/) if desired.
```
Comment (KyleAMathews) [inline docs/docs/how-to/local-development/javascript-tooling.md:9]: link out to home pages for the different tech

File: docs/docs/how-to/local-development/javascript-tooling.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+## Bundling and transpilation
+
+Gatsby uses webpack to bundle files and Babel to transpile Javascript.
+
+Gatsby ships with a default configuration that should work well for most projects, but if you want to modify it to load dependencies differently, use custom relative paths, include specific polyfills, or other customizations, you can [add custom webpack config](/docs/how-to/custom-configuration/add-custom-webpack-config/) and/or [use a custom .babelrc](/docs/how-to/custom-configuration/babel/)
```
Comment (KyleAMathews) [inline docs/docs/how-to/local-development/javascript-tooling.md:15]: mention that there are many gatsby plugins which handle webpack changes — often it's easier to use a community-support webpack modification than maintaining your own

File: docs/docs/how-to/local-development/javascript-tooling.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Gatsby offers support for the standard Javascript toolchain.
+
+## Typescript
+
+Gatsby supports Typescript automatically, with the [ability to modify the default configuration](gatsby-plugin-typescript/) if desired.
+
+## Bundling and transpilation
+
+Gatsby uses webpack to bundle files and Babel to transpile Javascript.
```
Comment (KyleAMathews) [inline docs/docs/how-to/local-development/javascript-tooling.md:13]: Just have bullets

- how to configure webpack
- how to configure Babel

File: docs/docs/how-to/local-development/javascript-tooling.md (hunk 2 above)
Comment (calcsam) [inline docs/docs/how-to/local-development/javascript-tooling.md:15]: * How to configure webapck [link]
* How to configure babel

[Note about plugins already exist for common things]

File: docs/docs/how-to/routing/creating-pages.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Gatsby supports multiple templated pages based on a single component. For example, a file located at `src/pages/products/{Product.name}.js` can generate pages like `/products/burger`, based on information coming in from a CMS or other data source. For details, look at the [File System Route API](/docs/reference/routing/file-system-route-api) documentation.
+
+## Creating pages from Markdown
+
+In order to enable a better content composition experience, Gatsby allows you to create both individual pages and dynamic routes using either [Markdown files](https://www.gatsbyjs.com/docs/how-to/routing/adding-markdown-pages/) or [MDX files](https://www.gatsbyjs.com/docs/how-to/routing/mdx/).
+
+## Using gatsby-node.js
+
+For more fine-grained control over routing, the `gatsby-node.js` file offers helpers that provide an escape hatch. More details in the [Routing Reference](https://www.gatsbyjs.com/docs/reference/routing/creating-routes/#using-gatsby-nodejs)
```
Comment (KyleAMathews) [inline docs/docs/how-to/routing/creating-pages.md:21]: "offers helpers that provide an escape hatch" is kinda awkward — maybe just — "you can programmatically create pages in `gatsby-node.js`"

File: docs/docs/how-to/routing/creating-pages.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+## Authentication gating routes
```
Comment (KyleAMathews) [inline docs/docs/how-to/routing/creating-pages.md:23]: just "Authenticated routes"

File: docs/docs/how-to/sourcing-data/headless-cms.md [hunk 6]
```diff
@@ -6,52 +6,54 @@ A _headless CMS_ is content management software that enables writers to produce
...
-| [Flotiq](https://flotiq.com/)                 | [guide](/docs/sourcing-from-flotiq/)                                             | [docs](/plugins/gatsby-source-flotiq)               | [starter](https://github.com/flotiq/gatsby-starter-blog)             |
-| [Tina](https://tina.io)                       | [guide](https://tina.io/guides/#gatsby)                                          | [docs](https://tina.io/docs/integrations/gatsby/)   |                                                                      |
-
-## How to add new guides to this section
-
-If you don’t see your preferred CMS in this list, you can [write a new guide yourself](/contributing/how-to-contribute/) or [open an issue to request it](https://github.com/gatsbyjs/gatsby/issues/new/choose).
-
-You can also [write your own source plugin](/docs/how-to/plugins-and-themes/creating-a-source-plugin/) to integrate Gatsby with a CMS that is not in the list.
+## Main CMSs
```
Comment (KyleAMathews) [inline docs/docs/how-to/sourcing-data/headless-cms.md:19]: "Main" isn't maybe the word we want? Maybe just "Popular CMSs"? Call it what it is 🤷

**Review by KyleAMathews — COMMENTED** [review]
🎉

File: docs/docs/how-to/local-development/javascript-tooling.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+Gatsby supports [Typescript](https://www.typescriptlang.org/) automatically, with the [ability to modify the default configuration](/plugins/gatsby-plugin-typescript/) if desired.
+
+## Bundling and transpilation
+
+Gatsby uses [webpack](https://webpack.js.org/) to bundle files and [Babel](https://babeljs.io/) to transpile Javascript.
+
+- [How to configure webpack](/docs/how-to/custom-configuration/add-custom-webpack-config/)
+- [How to configure Babel](/docs/how-to/custom-configuration/babel/)
+
+A number of Gatsby plugins already exist for relatively common tasks; you can install these in your `gatsby-config.js`.
```
Comment (KyleAMathews) [inline docs/docs/how-to/local-development/javascript-tooling.md:18]:
```suggestion
Most sites work great with the default configuration. Many common changes have [Gatsby plugins](/plugins) already which you can install in your `gatsby-config.js`.
```

File: docs/docs/how-to/routing/creating-pages.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+## Creating pages from Markdown
+
+In order to enable a better content composition experience, Gatsby allows you to create both individual pages and dynamic routes using either [Markdown files](https://www.gatsbyjs.com/docs/how-to/routing/adding-markdown-pages/) or [MDX files](https://www.gatsbyjs.com/docs/how-to/routing/mdx/).
+
+## Using gatsby-node.js
+
+If you need more fine-grained control over routing, you can programmatically create pages in `gatsby-node.js`. More details in the [Routing Reference](https://www.gatsbyjs.com/docs/reference/routing/creating-routes/#using-gatsby-nodejs)
+
+## Authenticated routes
```
Comment (KyleAMathews) [inline docs/docs/how-to/routing/creating-pages.md:23]:
```suggestion
## Dynamic and Authenticated routing
```

**Review by KyleAMathews — APPROVED** [review]
:shipit:

**Pull request merged**